.circleci/config.yml

```diff
@@ -0,0 +1,201 @@
+      - website/node_modules
+    key: v1-website-dependencies-{{ .Branch }}-{{ checksum "website/package.json" }}
+
+  - &restore-cache-danger
+    keys:
+      - v1-danger-dependencies-{{ .Branch }}-{{ checksum "danger/package.json" }}
+      # Fallback in case checksum fails
+      - v1-danger-dependencies-{{ .Branch }}-
+  - &save-cache-danger
+    paths:
+      - danger/node_modules
+    key: v1-danger-dependencies-{{ .Branch }}-{{ checksum "danger/package.json" }}
+
+  - &restore-cache-android
+    keys:
+      - v1-android-dependencies-{{ .Branch }}-{{ checksum "ReactAndroid/build.gradle" }}
+      # Fallback in case checksum fails
+      - v1-android-dependencies-{{ .Branch }}-
+  - &save-cache-android
+    paths:
+     - "ReactAndroid/build/downloads"
+    key: v1-android-dependencies-{{ .Branch }}-{{ checksum "ReactAndroid/build.gradle" }}
+
+  - &restore-cache-buck
+    keys:
+# TODO: Pick file to checksum against
+      - v1-buck-dependencies-{{ .Branch }}
+  - &save-cache-buck
+    paths:
+     - "/home/circleci/buck"
+    key: v1-buck-dependencies-{{ .Branch }}
+
+version: 2
+jobs:
+  test-node-8:
+    working_directory: ~/react-native
+    docker:
+      - image: circleci/node:8
+    steps:
+      - checkout
+      - run: npm install --no-package-lock
+      - run: |
+          npm run lint
+          npm run flow -- check
+          npm test -- --maxWorkers=1
+
+  test-node-6:
+    working_directory: ~/react-native
+    docker:
+      - image: circleci/node:6.11.0
+    steps:
+      - checkout
+      - run: npm install --no-package-lock
+      - run: |
+          npm run lint
+          npm run flow -- check
+          npm test -- --maxWorkers=1
+
+  test-node-4:
+    working_directory: ~/react-native
+    docker:
+      - image: circleci/node:4.8.4
+    steps:
+      - checkout
+      - run: npm install --no-package-lock
+      - run: |
+          npm run lint
+          npm run flow -- check
+          npm test -- --maxWorkers=1
+
+  build-js-bundle:
+    working_directory: ~/react-native
+    docker:
+      - image: circleci/node:8
+    steps:
+      - checkout
+      - run: npm install --no-package-lock
+      - run:
+          name: Building JS bundle for instrumentation tests
+          command: node local-cli/cli.js bundle --max-workers 1 --platform android --dev true --entry-file ReactAndroid/src/androidTest/js/TestBundle.js --bundle-output ReactAndroid/src/androidTest/assets/AndroidTestBundle.js
+      - persist_to_workspace:
+          root: ReactAndroid/src/androidTest/assets/
+          paths:
+            - AndroidTestBundle.js
+
+  test-android:
+    working_directory: ~/react-native
+    docker:
+      - image: circleci/openjdk:8-jdk
+    environment:
+      - TERM: "dumb"
+      - ADB_INSTALL_TIMEOUT: 10
+      - GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xmx512m -XX:+HeapDumpOnOutOfMemoryError"'
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ReactAndroid/src/androidTest/assets/
+      - run: |
```

Comment:
Maybe give this step a name so that CircleCI shows the name instead of this whole command in its web UI

Reply:
Hadn't seen this comment but I believe it's done now :)